src/instructlab/provenance/sigstore.py

```diff
@@ -0,0 +1,54 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+from pathlib import Path
+
+# Third-party
+from sigstore.sign import SigningContext
+from sigstore.oidc import Issuer
+from sigstore.models import Bundle
+from sigstore.verify import Verifier
+from sigstore.verify.policy import Identity
+
+def sign(model_path, bundle_path, staging=False):
```

Comment:
I recommend to use pathlib for input and output

```suggestion
def sign_model(model_path: pathlib.Path, bundle_path: pathlib.Path, staging: bool=False) -> None:
```

Reply:
Done, thanks for the recommendation.